Std/Data/List/Basic.lean

```diff
@@ -823,6 +823,40 @@ inductive Forall₂ (R : α → β → Prop) : List α → List β → Prop
 
 attribute [simp] Forall₂.nil
 
+/--
+Decision procedure for `Forall₂ R l₁ l₂`
+-/
+def Forall₂.dec (R : α → β → Prop) [∀ a b, Decidable (R a b)] : List α → List β → Bool
+  | [], [] => true
+  | a::as, b::bs =>
+    if R a b then
+      dec R as bs
+    else false
+  | _, _ => false
+
+theorem Forall₂.dec_iff {R : a → b → Prop} [∀ a b, Decidable (R a b)] :
+    ∀ l₁ l₂, Forall₂.dec R l₁ l₂ ↔ Forall₂ R l₁ l₂
+  | [], [] => by simp [dec]
+  | a::as, b::bs => by
+    if h : R a b then
+      simp [dec, h, dec_iff]
+      constructor <;> intro h'
+      exact Forall₂.cons h h'
+      have .cons _ tail := h'
+      exact tail
+    else
+      simp [dec, h]
+      intro h'
+      have .cons head _ := h'
+      exact h head
+  | _::_, [] | [], _::_ => by
+    simp [dec]
+    intro h
+    nomatch h
+
+instance {R : α → β → Prop} [∀ a b, Decidable (R a b)] : ∀ l₁ l₂, Decidable (Forall₂ R l₁ l₂) :=
```

Comment:
```suggestion
instance {R : α → β → Prop}  [DecidableRel R] : DecidableRel (Forall₂ R) :=
```

Reply:
Oh nevermind

Reply:
doesn't work because a and b are different types